include/albatross/src/core/model.hpp

```diff
@@ -28,7 +28,13 @@ template <typename ModelType> class ModelBase : public ParameterHandlingMixin {
   template <typename T, typename FeatureType> friend class fit_model_type;
 
 protected:
-  ModelBase() : insights(), use_async_(DEFAULT_USE_ASYNC){};
+  ModelBase()
+      : insights(),
+        threads_(DEFAULT_USE_ASYNC
+                     ? std::make_shared<ThreadPool>(std::max(
```

Comment:
should this use `get_default_thread_count`?

Reply:
I could do that, but it would change the `#include` structure and include the async utils into `core/`.  If that's OK, I can do it; otherwise I thought I'd write it out to avoid changing the graph of inclusions.

Reply:
I see. I guess the alternative would be to add a separate header which basically just holds this and include it in both core and async tooling?

Anyway, the include dependencies are far from perfect, no need to change this if you don't think it's worth it.